sqlglot/dialects/mysql.py

```diff
@@ -699,6 +700,9 @@ class Generator(generator.Generator):
         TRANSFORMS = {
             **generator.Generator.TRANSFORMS,
             exp.ArrayAgg: rename_func("GROUP_CONCAT"),
+            exp.AtTimeZone: lambda self, e: self.func(
+                "CONVERT_TZ", e.this, exp.var("@@global.time_zone"), e.args.get("zone")
+            ),
```

Comment:
I'm not sure if this assumption is safe. It's better to call `self.unsupported` and avoid generating something wrong vs silently corrupting data if that's the tradeoff we're looking at. Or, alternatively, you could maybe try configuring this using a dialect-level setting, thus opening this transpilation path conditionally.

Reply:
I think you're right. The problem I was trying to solve was constructing a "generic" AST using SQLGlot directly and then having it render appropriately across the database engines supported by SQLMesh. 

`exp.AtTimeZone` works for every dialect except MySQL, so I wanted to see if it could be made to work without introducing special hacks to replace that AST node with something else just for MySQL.

But I think if the concept isn't directly translatable, then you're right, it's better to raise `unsupported` to prevent surprises

Reply:
Actually, I notice the Spark dialect transpiles `exp.AtTimeZone` to `FROM_UTC_TIMESTAMP`. Isn't that making an assumption about the source timezone?

Reply:
Ah, maybe, need to double check that but it seems like it was [added a long time ago](https://github.com/tobymao/sqlglot/commit/80401ddcfe22b13922a51aaf0d1a556af246a277). Perhaps we should fix it while we're at it if it's problematic?

cc @tobymao

Reply:
sure if it's wrong we should fix it

Reply:
Hmm, I dug deeper into the Spark one at the reason it generates `exp.AtTimeZone` to `FROM_UTC_TIMESTAMP` is because it _parses_ `FROM_UTC_TIMESTAMP`  into `exp.AtTimeZone`.

I took a quick look at parsing `FROM_UTC_TIMESTAMP` into `exp.FromTimeZone` instead, but then I noticed that `TO_UTC_TIMESTAMP` is parsed into `exp.FromTimeZone` and `exp.FromTimeZone` is generated to `TO_UTC_TIMESTAMP`. So if I parsed `FROM_UTC_TIMESTAMP` into `exp.FromTimeZone`, it would be generated as  `TO_UTC_TIMESTAMP` which is wrong.

In addition, I did some tests around what Spark does with `TIMESTAMP` values and compared it to Postgres. As a baseline, I focused on the following:

```
parse_one("select '2023-01-01 12:13:14-08:00'::timestamp at time zone 'UTC'", dialect="postgres").sql(dialect="spark")
"SELECT FROM_UTC_TIMESTAMP(CAST('2023-01-01 12:13:14-08:00' AS TIMESTAMP), 'UTC')"
```

On Postgres, the returned values are consistent regardless of the session timezone:
```
set time zone 'UTC';
select '2023-01-01 12:13:14-08:00'::timestamp at time zone 'UTC'; --2023-01-01 12:13:14.000000 +0000

set time zone 'America/Los_Angeles';
select '2023-01-01 12:13:14-08:00'::timestamp at time zone 'UTC'; --2023-01-01 12:13:14.000000 +0000
```

On Spark, the returned values vary depending on the session timezone:
```
set time zone 'UTC';
SELECT FROM_UTC_TIMESTAMP(CAST('2023-01-01 12:13:14-08:00' AS TIMESTAMP), 'UTC') --2023-01-01 20:13:14.000000

set time zone 'America/Los_Angeles';
SELECT FROM_UTC_TIMESTAMP(CAST('2023-01-01 12:13:14-08:00' AS TIMESTAMP), 'UTC') --2023-01-01 12:13:14.000000
```

The transpilation gives different results because how the `CAST` on the spark side is interpreted depends on the session timezone.

Since it's not a straightforward fix, perhaps it can be addressed in a future PR?

Reply:
For MySQL, i've updated it to emit:
```
self.unsupported("AT TIME ZONE is not supported by MySQL")
```
if the generator encounters an `exp.AtTimeZone` node

Reply:
Sounds good 👍